codice/ddf
# DDF-4024: Export results component (UI).

Added react component that allows exporting result sets in various formats (dictated by available query response transformers present).

#### What does this PR do?
#### Who is reviewing it? 

@jrnorth 
@brianfelix 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@djblue 
@rzwiefel 
#### How should this be tested?
1. Search
<img width="1616" alt="screen shot 2018-09-05 at 10 29 30 am" src="https://user-images.githubusercontent.com/118054/45100263-2214cf80-b0f7-11e8-963a-f7c947ffacc3.png">
2. Add Visual -> Table
<img width="1616" alt="screen shot 2018-09-05 at 10 30 12 am" src="https://user-images.githubusercontent.com/118054/45100309-3789f980-b0f7-11e8-8cc4-909be650037f.png">
- Should see the component above table on the right
- Should have a drop-down for 'visible' and 'all' followed by drop-down containing available 
transformers (by their service id)
<img width="140" alt="screen shot 2018-09-05 at 10 29 42 am" src="https://user-images.githubusercontent.com/118054/45100334-41abf800-b0f7-11e8-8493-725c5a589c6e.png">
<img width="263" alt="screen shot 2018-09-05 at 10 29 48 am" src="https://user-images.githubusercontent.com/118054/45100335-42448e80-b0f7-11e8-91ab-5874fd52888b.png">
3. Select 'visible' or 'all'
4. Select the transformer to perform the export
5. Click download icon
<img width="1616" alt="screen shot 2018-09-05 at 10 30 00 am" src="https://user-images.githubusercontent.com/118054/45100431-6e600f80-b0f7-11e8-820a-80a4ba865105.png">
6. You should have a file corresponding to the transformer selected

#### Any background context you want to provide?
This story focuses on exposing the available transformers as a ui component and utilizing them to perform the work (transformation). Each transformer is responsible for performing the transformation correctly and should be tested/verified individually.
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-4024
#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
